core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -387,6 +362,31 @@ private[deploy] class Master(
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
+    case RegisterWorker(
+        id, workerHost, workerPort, workerRef, cores, memory, workerUiPort, publicAddress) => {
+      logInfo("Registering worker %s:%d with %d cores, %s RAM".format(
+        workerHost, workerPort, cores, Utils.megabytesToString(memory)))
+      if (state == RecoveryState.STANDBY) {
+        // ignore, don't send response
```

Comment:
what I'm saying is in general in `receiveAndReply` you should always reply because that's what the caller expects. Otherwise it's really confusing when we get future timeouts cause they're hard to debug.